src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -187,7 +230,8 @@ def update_stats(self, area, simulation_status, progress_info):
 
         self.market_unmatched_loads.update_unmatched_loads(area)
 
-        self.device_statistics.update(area)
+        self.device_statistics.update(area, self.area_result_dict,
+                                      self.flattened_area_core_stats_dict)
```

Comment:
I thought the goal of this story is to get rid of parsing the area object.
Do you really need it here? IMO it should look like:
```suggestion
        self.device_statistics.update(self.area_result_dict,
                                      self.flattened_area_core_stats_dict)
```
Right?

Reply:
Yes you are right, already got rid of using the area object inside the `update` method, will now also change this method signature too.